src/Stratis.FederatedPeg.Features.FederationGateway/MonitorChain/OpReturnDataReader.cs

```diff
@@ -142,5 +139,28 @@ public string TryGetTargetAddressFromOpReturn(Transaction transaction)
 
             return opReturnAddresses.Count != 1 ? null : opReturnAddresses[0];
         }
+
+        /// <inheritdoc />
+        public string TryGetTransactionIdFromOpReturn(Transaction transaction)
+        {
+            var transactionId = SelectBytesContentFromOpReturn(transaction)
+                .Select(this.TryConvertValidOpReturnDataToHash)
+                .Where(s => s != null)
+                .Distinct(StringComparer.InvariantCultureIgnoreCase).ToList();
+
+            this.logger.LogDebug("TransactionId found in OP_RETURN(s) of transaction {0}: [{1}]",
+                transaction.GetHash(), string.Join(",", transactionId));
+

```

Comment:
string.Join happens on transactionId, which suggests there could be more than one.  If so maybe the log should read "TransactionId(s)...".